Ginger/Ginger/ApplicationModelsLib/ModelParams Pages/ModelParamsPage.xaml.cs

```diff
@@ -142,12 +144,26 @@ private void InitModelParametersGrid()
 
             ModelParametersGrid.SetbtnDeleteHandler(new RoutedEventHandler(DeleteParams_Clicked));
             ModelParametersGrid.SetbtnClearAllHandler(new RoutedEventHandler(ClearAllParams_Clicked));
+            ModelParametersGrid.AddToolbarTool(eImageType.ExcelFile, "Export Optional Values For Parameters", new RoutedEventHandler(ExportOptionalValuesForParameters));
         }
         private void ImportOptionalValuesForParameters(object sender, RoutedEventArgs e)
         {            
             WizardWindow.ShowWizard(new AddModelOptionalValuesWizard((ApplicationModelBase)mApplicationModel));
             ModelParametersGrid.DataSourceList = mApplicationModel.AppModelParameters;
         }
+
+        private void ExportOptionalValuesForParameters(object sender, RoutedEventArgs e)
+        {
+            ImportOptionalValuesForParameters im = new ImportOptionalValuesForParameters();
+            List<AppParameters> parameters = new List<AppParameters>();
+            foreach (var prms in mApplicationModel.AppModelParameters)
+            {
+                im.AddNewParameterToList(parameters, prms);
+            }
+            string filePath = im.ExportParametersToExcelFile(parameters, string.Format("{0}_Parameters", mApplicationModel.Name));
+            Process.Start(filePath);
+        }
+
         private void UploadToGlobalParam(object sender, RoutedEventArgs e)
```

Comment:
"ExportParametersToExcelFile" is already separated but the other code mostly dependent on local object so it might not be that helpful if we try to combine it.